### PR TITLE
refactor: enhanced createApiResponseDto and implemented

### PR DESCRIPTION
**Pull request by esgiraldop**

### CODE REVIEW

**Review by zarksoad — APPROVED** [review]

File: src/common/helpers/create-api-response-dto.helper.ts
Comment (daguttt) [inline src/common/helpers/create-api-response-dto.helper.ts:1]: Nice!

File: src/modules/user/services/get-latest-classes-inprogress-byCourse-byUser.service.ts [hunk 1]
```diff
@@ -3,55 +3,81 @@ import { Repository } from "typeorm";
...
         @InjectRepository(UserClass) private userClassRepository: Repository<UserClass>,
-        private findUserByIdService:FindUserByIdService
-    ){}
+        private findUserByIdService: FindUserByIdService,
+        private courseService: CourseService
+    ) { }
 
-    async getLatestClassesInProgressByUserByCourse(id:number):Promise<GetLatestClassesInProgressByCourseByUserResponseDto[]>
+    async getLatestClassesInProgressByUserByCourse(id: number): Promise<GetLatestClassesInProgressByCourseByUserResponseDto[]>
     // Promise<SectionClass>{
```
Comment (daguttt) [inline src/modules/user/services/get-latest-classes-inprogress-byCourse-byUser.service.ts]: Remove this comment

`// Promise<SectionClass>`

**Review by daguttt — CHANGES_REQUESTED** [review]

File: src/modules/user/services/get-latest-classes-inprogress-byCourse-byUser.service.ts (hunk 1 above)
Comment (esgiraldop) [reply src/modules/user/services/get-latest-classes-inprogress-byCourse-byUser.service.ts]: Comment removed

**Review by christian75876 — APPROVED** [review]
todo muy bien

**Pull request merged**